invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,73 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        :param identity: The user identity performing the operation.
+        :param data: A list of tuples containing the record ID and record data.
+        :param uow: The unit of work to register the record operations. Defaults to None.
+
+        Returns:
+            list: A list of tuples containing the operation type ('create' or 'update'), the processed record or the record dict, and any schema errors encountered.
+        """
+        records_processed = []
+        for record_id, record_dict in data:
+            try:
+                record = self.record_cls.pid.resolve(record_id)
+
+                # Permissions
+                self.require_permission(identity, "update", record=record)
+                record_data, schema_errors = self.schema.load(
+                    record_dict,
+                    context=dict(identity=identity, pid=record.pid, record=record),
+                    raise_errors=False,
+                )
+
+                # If errors we avoid creating the record
+                if schema_errors:
+                    records_processed.append(("update", record_dict, schema_errors))
+                    continue
+
+                # Run components
+                self.run_components(
+                    "update", identity, data=record_data, record=record, uow=uow
+                )
+
+                records_processed.append(("update", record, schema_errors))
+            except (NoResultFound, PIDDoesNotExistError):
```

Comment:
I am curious why sometimes we can get NoResultFound and PIDDoesNotExistError, does it mean that we mix the way of accessing the record?

Reply:
Right, not sure why I added `NoResultFound`, I will test it again but should rely only on `PIDDoesNotExistError`

Reply:
Some vocabularies don't rely on `PersistentIdentifier` in which case they don't raise a `PIDDoesNotExistError`, but instead `NoResultFound` (since they do a direct query to the DB on the table that holds the vocabulary).